terraform/examples/io500/stop.sh

```diff
@@ -3,15 +3,68 @@
 set -e
 trap 'echo "Hit an unexpected and unchecked error. Exiting."' ERR
 
-# Load needed variables
-source ./configure.sh
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
 
-printf "
--------------------------------------------------------------------------------
-Destroying DAOS Servers & Clients
--------------------------------------------------------------------------------
-"
+# Directory where all generated files were stored
+IO500_TMP="${SCRIPT_DIR}/tmp"
 
-pushd ../full_cluster_setup
+# Directory containing config files
+CONFIG_DIR="${CONFIG_DIR:-${SCRIPT_DIR}/config}"
+
+# Config file in ./config that is used to spin up the environment and configure IO500
+CONFIG_FILE="${CONFIG_FILE:-config.sh}"
+
+# active_config.sh is a symlink to the last config file used by start.sh
+ACTIVE_CONFIG="${CONFIG_DIR}/active_config.sh"
+
+log() {
+  msg="$1"
+  print_lines="$2"
+  # shellcheck disable=SC2155,SC2183
+  local line=$(printf "%80s" | tr " " "-")
+  if [[ -t 1 ]]; then tput setaf 14; fi
+  if [[ "${print_lines}" == 1 ]]; then
+    printf -- "\n%s\n %-78s \n%s\n" "${line}" "${msg}" "${line}"
+  else
+    printf -- "\n%s\n\n" "${msg}"
+  fi
+  if [[ -t 1 ]]; then tput sgr0; fi
+}
+
+log_error() {
+  # shellcheck disable=SC2155,SC2183
+  if [[ -t 1 ]]; then tput setaf 160; fi
+  printf -- "\n%s\n\n" "${1}" >&2;
+  if [[ -t 1 ]]; then tput sgr0; fi
+}
+
+log_section() {
+  log "$1" "1"
+}
+
+# Source the active config file that was last used by ./start.sh
+if [[ ! -L "${ACTIVE_CONFIG}" ]]; then
+  log_error "'${ACTIVE_CONFIG}' symlink does not exist."
+  log_error "Either the start.sh script was never run or it did not run successfully."
+  log_error "Unable to perform a terraform destroy. Exiting."
+  exit 1
+fi
+
+# Source the last configuration that was used by the start.sh script
+source "$(readlink ${ACTIVE_CONFIG})"
+
+log_section "Destroying DAOS Servers & Clients"
+
+pushd "${SCRIPT_DIR}/../full_cluster_setup"
 terraform destroy -auto-approve
```

Comment:
@markaolson  can you help me understand all the ACTIVE_CONFIG logic?   

What is the use case?

Reply:
The config/active_config.sh symlink is created by the start.sh script. It points to the config that was sourced when the start script ran terraform to provision the instances.  The symlink is then used by the stop.sh script to know which config to source when it runs terraform to tear down the environment.  Without the symlink you you have to manually pass the last config you used to the stop.sh script every time you run it.

Reply:
stop and start should just be + ansible start + ansible destroy.

What happens if I spawn 2 different DAOS system for 2 difference clusters from the same project?   

After I create the 2nd how do a I get back to the 1st?